scripts/apps/authoring-react/toolbar/highlights-modal.tsx

```diff
@@ -0,0 +1,114 @@
+import {sdApi} from 'api';
+import {httpRequestJsonLocal} from 'core/helpers/network';
+import {dispatchInternalEvent} from 'core/internal-events';
+import {Spacer} from 'core/ui/components/Spacer';
+import {gettext} from 'core/utils';
+import {isEqual} from 'lodash';
+import React from 'react';
+import {IArticle, IHighlight, IHighlightResponse} from 'superdesk-api';
+import {Button, Modal, MultiSelect} from 'superdesk-ui-framework/react';
+
+interface IProps {
+    closeModal(): void;
+    article: IArticle;
+}
+
+interface IStateLoading {
+    initialized: false;
+}
+
+interface IStateLoaded {
+    initialized: true;
+    availableHighlights: Array<IHighlight> | null;
+    markedHighlights: Array<string> | null;
+    isSaving: boolean;
+}
+
+type IState = IStateLoaded | IStateLoading;
+
+export default class HighlightsModal extends React.PureComponent<IProps, IState> {
+    constructor(props: IProps) {
+        super(props);
+
+        this.state = {
+            initialized: false,
+        };
+
+        this.markHighlights = this.markHighlights.bind(this);
+    }
+
+    componentDidMount(): void {
+        Promise.all([
+            sdApi.highlights.fetchHighlights(),
+            this.fetchArticleWithHighlights(),
+        ]).then(([highlightResponse, article]: [IHighlightResponse, IArticle]) => {
+            this.setState({
+                initialized: true,
+                availableHighlights: highlightResponse._items,
+                markedHighlights: article.highlights,
+            });
+        });
+    }
+
+    markHighlights(): void {
+        if (this.state.initialized) {
+            this.setState({...this.state, isSaving: true});
+            sdApi.highlights.markItem(this.state.markedHighlights, this.props.article._id).then(() => {
+                this.props.closeModal();
+                dispatchInternalEvent('dangerouslyForceReloadAuthoring', undefined);
+            });
+        }
+    }
+
+    fetchArticleWithHighlights(): Promise<IArticle> {
```

Comment:
not bound, on the other hand, it doesn't use anything from the class so it shouldn't be in the class anyway. You should use `sdApi.article.get` instead